src/Ast/Type/TypeProjectionNode.php

```diff
@@ -0,0 +1,33 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\PhpDocParser\Ast\Type;
+
+use PHPStan\PhpDocParser\Ast\NodeAttributes;
+
+class TypeProjectionNode implements TypeNode
+{
+
+	use NodeAttributes;
+
+	/** @var TypeNode */
+	public $type;
+
+	/** @var 'covariant'|'contravariant' */
+	public $variance;
```

Comment:
Shouldn't be here also invariant variance and be the default?

Reply:
And maybe use class string constants instead of literal strings so it can be nicely checked in user code :)

Reply:
What about doing a similar thing I suggested in phpstan-src and make the variance simply an additional property of GenericTypeNode instead?

Reply:
@jiripudil sorry, you're right, I missed the 359's early return. Anyway, wouldn't you rather consider something like `GenericTypeParameterNode($type, $variance)` (and ~ `GenericStarParameterNode`) -- i.e. the parser describes what it parses, not what is the semantic meaning.

Reply:
(but that's BC break :/ if done through composition)